test_partner_client.py

```diff
@@ -0,0 +1,131 @@
+
+from datetime import datetime
+import requests
+from requests import ConnectionError, Timeout
+import retry
+import re
+
+# TODO Figure out logging given how Clarity works.
+
+
+class TestPartnerSampleInformation(object):
+
+    def __init__(self, referral_code, lab_referral, arrival_date, result_date, comment, cov19_result):
+
+        # REVIEW Discussion point. Not all values here are required, but I see no reason not
+        #        to send them, since I do thing we should have them for all cases.
+
+        super().__init__()
+
+        if not isinstance(str, referral_code) and not len(referral_code) == 10:
+            raise AssertionError(
+                "referral_code needs to be 10 digit number string.")
+
+        if not (int(referral_code[:-1]) % 10 == int(referral_code[-1])):
+            raise AssertionError("Check code digit {} (last digit) did not match mod10 requierment" +
```

Comment:
Isn't the check more complex than this? I'm assuming it's https://en.wikipedia.org/wiki/Luhn_algorithm

An argument could be made for that we don't have to check for this here, as it will be checked by the external provider, but I'm not against it either!

Reply:
Yes, I also realized this when I started writing tests. It's fixed now.